### DISCUSSION TITLE
Mesh repairing

### DISCUSSION THREAD

**Discussion by hbadi** [CGAL/cgal#6444] [Q&A] [answered]
Hi,

Is there some algorithm in cgal to repair a conforming 3d tetra mesh which contains degenerated and very small (less than 1E-10) volume element ?

Thanks for helping.

**Comment by janetournois** [comment, marked as answer]
Hi @hbadi 
what do you mean by "repair"?
[CGAL::tetrahedral_isotropic_remeshing()](https://doc.cgal.org/latest/Tetrahedral_remeshing/group__PkgTetrahedralRemeshingRef.html#gae4686577872290e0335439edafb27a7b) can improve its quality with respect to dihedral angles, if it is what you are looking for

**Reply by hbadi** [reply]
Hello @janetournois,

Yes using gmsh i get sometimes some degenerate tetras with negative or null measure. I want to clean the mesh with swapping faces/edges.

**Reply by janetournois** [reply]
Tetrahedral remeshing will not work with negative volumes, but null should be fine
